pkg/client/client.go

```diff
@@ -57,6 +57,153 @@ func (o *OptimizelyClient) CreateUserContext(userID string, attributes map[strin
+	if err != nil {
+		return NewErrorDecision(key, userContext, decide.GetDecideError(decide.FlagKeyInvalid, key))
+	}
+	decisionContext.Feature = &feature
+
+	usrContext := entities.UserContext{
+		ID:         userContext.GetUserID(),
+		Attributes: userContext.GetUserAttributes(),
+	}
+	var variationKey string
+	eventSent := false
+	flagEnabled := false
+	allOptions := o.getAllOptions(options)
+	decisionReasons := decide.NewDecisionReasons(options)
+	decisionContext.Variable = entities.Variable{}
+
+	featureDecision, _ := o.DecisionService.GetFeatureDecision(decisionContext, usrContext, options, decisionReasons)
+	if featureDecision.Variation != nil {
+		variationKey = featureDecision.Variation.Key
+		flagEnabled = featureDecision.Variation.FeatureEnabled
+	}
+
+	if !allOptions.DisableDecisionEvent {
+		if ue, ok := event.CreateImpressionUserEvent(decisionContext.ProjectConfig, featureDecision.Experiment,
+			featureDecision.Variation, usrContext, key, featureDecision.Experiment.Key, featureDecision.Source, flagEnabled); ok {
+			o.EventProcessor.ProcessEvent(ue)
+		}
+		eventSent = true
+	}
+
+	variableMap := map[string]interface{}{}
+	if !allOptions.ExcludeVariables {
+		variableMap = o.getDecisionVariableMap(feature, featureDecision.Variation, flagEnabled, decisionReasons)
+	}
+	optimizelyJSON := optimizelyjson.NewOptimizelyJSONfromMap(variableMap)
+	reasonsToReport := decisionReasons.ToReport()
+	ruleKey := featureDecision.Experiment.Key
+
+	if o.notificationCenter != nil {
+		decisionNotification := decision.FlagNotification(key, variationKey, ruleKey, flagEnabled, eventSent, usrContext, variableMap, reasonsToReport)
+		o.logger.Info(fmt.Sprintf(`Feature "%s" is enabled for user "%s"? %v`, key, usrContext.ID, flagEnabled))
+		if e := o.notificationCenter.Send(notification.Decision, *decisionNotification); e != nil {
+			o.logger.Warning("Problem with sending notification")
+		}
+	}
+
+	return NewOptimizelyDecision(variationKey, ruleKey, key, flagEnabled, optimizelyJSON, userContext, reasonsToReport)
+}
+
+func (o *OptimizelyClient) decideForKeys(userContext OptimizelyUserContext, keys []string, options decide.OptimizelyDecideOptions) map[string]OptimizelyDecision {
+	var err error
+	defer func() {
+		if r := recover(); r != nil {
+			switch t := r.(type) {
+			case error:
+				err = t
+			case string:
+				err = errors.New(t)
+			default:
+				err = errors.New("unexpected error")
+			}
+			errorMessage := fmt.Sprintf("decideForKeys call, optimizely SDK is panicking with the error:")
+			o.logger.Error(errorMessage, err)
+			o.logger.Debug(string(debug.Stack()))
+		}
+	}()
+
+	decisionMap := map[string]OptimizelyDecision{}
+	if _, err = o.getProjectConfig(); err != nil {
+		o.logger.Error("Optimizely instance is not valid, failing decideForKeys call.", err)
+		return decisionMap
+	}
+
+	if len(keys) == 0 {
+		return decisionMap
+	}
+
+	enabledFlagsOnly := o.getAllOptions(options).EnabledFlagsOnly
```

Comment:
Since `decideForKeys` calls `decide` internally and `decide` itself calls `allOptions` method, we are only using `allOptions` here to check `EnabledFlagsOnly` bool.